pkg/catalog/types.go

```diff
@@ -0,0 +1,48 @@
+package catalog
+
+import (
+	imagev1 "github.com/openshift/api/image/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ComponentType ...
+type ComponentType struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ComponentSpec `json:"spec,omitempty"`
+}
+
+// ComponentSpec ...
+type ComponentSpec struct {
+	Namespace      string              `json:"namespace"`
+	AllTags        []string            `json:"allTags"`
+	NonHiddenTags  []string            `json:"nonHiddenTags"`
+	ImageStreamRef imagev1.ImageStream `json:"imageStreamRef"`
+}
+
+// ComponentTypeList ...
+type ComponentTypeList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Items             []ComponentType `json:"items"`
+}
+
+// ServiceType ...
+type ServiceType struct {
```

Comment:
Same as `ComponentType`. Why not just `Service`?

Reply:
also about the `// ServiceType ...` would having more info in the comment help?

Reply:
I discussed with @kadel and ServiceType and Component Type is a better fit than just Service and Component

Reply:
more info in the comment?